OpenTrons/opentrons-api
# feat(components, protocol-designer): add Thermocycler step details UI

# Overview

Similar to viewing substeps detail for moveLiquid type steps, this PR adds UI for thermocycler substep details. Also, I move the step details toolbox to the right side of the screen as designs specify, and make a few other small styling changes to our ListItem, Toolbox, and StepOverFlowMenu components.

Closes AUTH-882

## Test Plan and Hands on Testing

- create or import a protocol ([example](https://github.com/user-attachments/files/17478058/fullPdWithTC.json.zip)) that contains a Thermocycler profile step
- on the step's overflow menu, select 'View details'
- verify that the step details correctly populate in a toolbox on the right side of the app

https://github.com/user-attachments/assets/ab5d44e9-17dd-42c5-bb67-e64c7766f0cb


## Changelog

- wire up Thermocycler substeps content
- move the substeps toolbox to the right of the screen
- add small style changes to `ListItem` padding and `Toolbox` width/padding
- add tests

## Review requests

- see test plan

## Risk assessment

low
